src/huggingface_hub/inference/_providers/fal_ai/text_to_image.py

```diff
@@ -0,0 +1,44 @@
+import json
+from typing import Any, Dict, Optional, Union
+
+from huggingface_hub.utils import get_session
+
+
+BASE_URL = "https://fal.run"
+
+SUPPORTED_MODELS = {
+    "black-forest-labs/FLUX.1-schnell": "fal-ai/flux/schnell",
+    "black-forest-labs/FLUX.1-dev": "fal-ai/flux/dev",
+}
+
+
+def build_url(model: Optional[str] = None) -> str:
+    return f"{BASE_URL}/{model}"
+
+
+def map_model(model: str) -> str:
+    mapped_model = SUPPORTED_MODELS.get(model)
+    if mapped_model is None:
+        raise ValueError(f"Model {model} is not supported for Fal.AI text-to-image task")
+    return mapped_model
+
+
+def prepare_headers(headers: Dict, **kwargs) -> Dict:
+    headers["Authorization"] = f"Key {kwargs['token']}"
+    return headers
```

Comment:
```suggestion
def prepare_headers(headers: Dict, *, token: Optional[str] = None) -> Dict:
    return {
        **headers,
        "Authorization": f"Key {token}"
    }
```

Let's not mutate `headers` in-place for safety